app/src/main/java/com/example/socialmediavbsanalay/data/dataSource/post/PostDataSource.kt

```diff
@@ -1,7 +1,11 @@
 package com.example.socialmediavbsanalay.data.dataSource.post
 
+import android.net.Uri
 import com.example.socialmediavbsanalay.domain.model.Post
+import kotlinx.coroutines.flow.Flow
 
 interface PostDataSource {
     suspend fun fetchPosts(): Result<List<Post>>
+    suspend fun uploadPhoto(imageUri: Uri): Unit
+    fun getPosts(): Flow<List<Post>>
```

Comment:
burada flow yerine suspend kullanabilir miydik? eğer kullanabilirsek bize ne tür artısı ya da eksisi olurdu? Açıklayabilir misin rica etsem

Reply:
flow zaten asenkron bir yapı ve reaktif buda daha büyük verilerin hızlı şekilmesinde bize fayda sağlıyor auth işlemlerinde mesela suspend kullandık çünkü çekilecek veri büyük değil flow kullanımına ihtiyaç yok ancak realtime ve sürekli güncellenen veriler için flow kullanımı daha uygun